tests: Mysterious test failure in TestListObjectPartsDiskNotFound

<!--- Provide a general summary of the issue in the Title above -->
[Build logs from travis](https://github.com/minio/minio/files/914120/log.txt)

## Expected Behavior
No failure in test cases expected.
<!--- If you're describing a bug, tell us what should happen -->
<!--- If you're suggesting a change/improvement, tell us how it should work -->

## Current Behavior
```
Running cyclo: Done.
Running all coverage for minio: ok  	github.com/minio/minio	0.010s	coverage: 66.7% of statements
?   	github.com/minio/minio/browser	[no test files]
OK: 201 passed
--- FAIL: TestListObjectPartsDiskNotFound (0.19s)
	object-api-multipart_test.go:1460: Test 5: XL: Expected to fail with error "Bucket not found: volatile-bucket-1", but instead failed with error "disk not found" instead
	object-api-multipart_test.go:1460: Test 6: XL: Expected to fail with error "Bucket not found: volatile-bucket-2", but instead failed with error "disk not found" instead
FAIL
coverage: 75.2% of statements
FAIL	github.com/minio/minio/cmd	81.275s
make: *** [coverage] Error 1
```
<!--- If describing a bug, tell us what happens instead of the expected behavior -->
<!--- If suggesting a change/improvement, explain the difference from current behavior -->

## Possible Solution
Do not know. 

<!--- Not obligatory, but suggest a fix/reason for the bug, -->
<!--- or ideas how to implement the addition or change -->

## Steps to Reproduce (for bugs)
Not reproducible easily requires `-race`

## Your Environment
Noticed in this PR - https://github.com/minio/minio/pull/4019

--- a/cmd/object-api-multipart_test.go
+++ b/cmd/object-api-multipart_test.go
@@ -1457,7 +1457,7 @@ func testListObjectPartsDiskNotFound(obj ObjectLayer, instanceType string, disks
 		// Failed as expected, but does it fail for the expected reason.
 		if actualErr != nil && !testCase.shouldPass {
 			if !strings.Contains(actualErr.Error(), testCase.expectedErr.Error()) {
-				t.Errorf("Test %d: %s: Expected to fail with error \"%s\", but instead failed with error \"%s\" instead", i+1, instanceType, testCase.expectedErr.Error(), actualErr.Error())
+				t.Errorf("Test %d: %s: Expected to fail with error \"%s\", but instead failed with error \"%s\" instead", i+1, instanceType, testCase.expectedErr, actualErr)
 			}
 		}
 		// Passes as expected, but asserting the results.

--- a/cmd/xl-v1-bucket.go
+++ b/cmd/xl-v1-bucket.go
@@ -113,27 +113,34 @@ func undoMakeBucket(storageDisks []StorageAPI, bucket string) {
 
 // getBucketInfo - returns the BucketInfo from one of the load balanced disks.
 func (xl xlObjects) getBucketInfo(bucketName string) (bucketInfo BucketInfo, err error) {
+	var bucketErrs []error
 	for _, disk := range xl.getLoadBalancedDisks() {
 		if disk == nil {
+			bucketErrs = append(bucketErrs, errDiskNotFound)
 			continue
 		}
-		var volInfo VolInfo
-		volInfo, err = disk.StatVol(bucketName)
-		if err == nil {
+		volInfo, serr := disk.StatVol(bucketName)
+		if serr == nil {
 			bucketInfo = BucketInfo{
 				Name:    volInfo.Name,
 				Created: volInfo.Created,
 			}
 			return bucketInfo, nil
 		}
-		err = traceError(err)
+		err = traceError(serr)
 		// For any reason disk went offline continue and pick the next one.
 		if isErrIgnored(err, bucketMetadataOpIgnoredErrs...) {
+			bucketErrs = append(bucketErrs, err)
 			continue
 		}
-		break
+		// Any error which cannot be ignored, we return quickly.
+		return BucketInfo{}, err
 	}
-	return BucketInfo{}, err
+	// If all our errors were ignored, then we try to
+	// reduce to one error based on read quorum.
+	// `nil` is deliberately passed for ignoredErrs
+	// because these errors were already ignored.
+	return BucketInfo{}, reduceReadQuorumErrs(bucketErrs, nil, xl.readQuorum)
 }
 
 // GetBucketInfo - returns BucketInfo for a bucket.
@@ -180,6 +187,7 @@ func (xl xlObjects) listBuckets() (bucketsInfo []BucketInfo, err error) {
 			}
 			return bucketsInfo, nil
 		}
+		err = traceError(err)
 		// Ignore any disks not found.
 		if isErrIgnored(err, bucketMetadataOpIgnoredErrs...) {
 			continue

